File: 02_Sort_Application/Level2/11651.cpp
// 좌표 정렬하기 2
// https://www.acmicpc.net/problem/11651
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

// x, y좌표계 정보를 담는 struct
struct Point
{
    int x, y;
};

// a < b일 때 참이 되도록 return하면 오름차순으로 정렬됨
bool cmp(Point a, Point b)
{
    // y좌표가 같을땐 x좌표로 오름차순 정렬
    if (a.y == b.y)
    {
        return a.x < b.x;
    }
    // y좌표가 다를땐 y좌표로 오름차순 정렬
    return a.y < b.y;
}

int main()
{
    int n;
    cin >> n;
    vector<Point> v(n);
    for (int i = 0; i < n; i++)
    {
        cin >> v[i].x >> v[i].y;
    }
    sort(v.begin(), v.end(), cmp);
    for (int i = 0; i < n; i++)
    {
        cout << v[i].x << " " << v[i].y << "\n";
    }
    return 0;
}

File: 02_Sort_Application/Level2/11651.py
# 좌표 정렬하기
# https://www.acmicpc.net/problem/11650
import sys

if __name__ == "__main__":
    N = int(input())
    coord_list = []

    for _ in range(N):
        # input() 사용시 시간 초과.
        x, y = map(int, sys.stdin.readline().split())
        coord_list.append([x, y])

    # y, x 순으로 정렬
    coord_list.sort(key=lambda x: (x[1], x[0]))

    # 출력
    for coord in coord_list:
        print(coord[0], coord[1])